### PR TITLE
Mon 4940 check poller ip address validity master

### PR DESCRIPTION
**Pull request by lpinsivy**
## Description

Check the validity of poller IP address

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

#### Community contributors & Centreon team

- [ ] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/class/centreon-clapi/centreonInstance.class.php [hunk 1]
```diff
@@ -126,6 +136,13 @@ public function initUpdateParameters($parameters)
             throw new CentreonClapiException(self::MISSINGPARAMETER);
         }
 
+        if ($params[1] == 'ns_ip_address' && !preg_match('/^([0-9a-fA-F]{4}|0)(\:([0-9a-fA-F]{4}|0)){7}$/', $params[2])
```
Comment (sc979) [inline www/class/centreon-clapi/centreonInstance.class.php]:
```suggestion
        // Check IPv6, IPv4 and FQDN format
        if ($params[1] == 'ns_ip_address' && !preg_match('/^([0-9a-fA-F]{4}|0)(\:([0-9a-fA-F]{4}|0)){7}$/', $params[2])
```

File: www/class/centreon-clapi/centreonInstance.class.php [hunk 2]
```diff
@@ -126,6 +136,14 @@ public function initUpdateParameters($parameters)
             throw new CentreonClapiException(self::MISSINGPARAMETER);
         }
 
+        // Check IPv6, IPv4 and FQDN format
+        if ($params[1] == 'ns_ip_address' && !preg_match('/^([0-9a-fA-F]{4}|0)(\:([0-9a-fA-F]{4}|0)){7}$/', $params[2])
```
Comment (kduret) [inline www/class/centreon-clapi/centreonInstance.class.php]: there are already FILTER_VALIDATE_DOMAIN and FILTER_VALIDATE_IP which can be used with `filter_var`

Comment (lpinsivy) [reply www/class/centreon-clapi/centreonInstance.class.php]: done

File: src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php [hunk 3]
```diff
@@ -327,10 +327,7 @@ public function postLinkCentreonRemoteServer(): array
         $serverName = substr($this->arguments['server_name'], 0, 40);
 
         // Check IPv6, IPv4 and FQDN format
-        if (!preg_match('/^([0-9a-fA-F]{4}|0)(\:([0-9a-fA-F]{4}|0)){7}$/', $serverIP)
-            && !(preg_match('/^[0-9\.]+$/', $serverIP) && preg_match('/^((25[0-5]|(2[0-4]|1[0-9]|[1-9]|)[0-9])(\.(?!$)|$)){4}$/', $serverIP))
-            && !(!preg_match('/^[0-9\.]+$/', $serverIP) && preg_match('/^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9]+)\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$/', $serverIP))
-        ) {
+        if (! filter_var($serverIP, FILTER_VALIDATE_DOMAIN) && !filter_var($serverIP, FILTER_VALIDATE_IP)) {
```
Comment (sc979) [inline src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php]:
```suggestion
        if (!filter_var($serverIP, FILTER_VALIDATE_DOMAIN) && !filter_var($serverIP, FILTER_VALIDATE_IP)) {
```

Comment (sc979) [reply src/CentreonRemote/Application/Webservice/CentreonConfigurationRemote.php]: typo correction

File: www/class/centreon-clapi/centreonInstance.class.php [hunk 4]
```diff
@@ -110,10 +110,9 @@ public function initInsertParameters($parameters)
         $addParams['ssh_port'] = $params[self::ORDER_SSH_PORT];
 
         // Check IPv6, IPv4 and FQDN format
-        if (!preg_match('/^([0-9a-fA-F]{4}|0)(\:([0-9a-fA-F]{4}|0)){7}$/', $addParams['ns_ip_address'])
-            && !(preg_match('/^[0-9\.]+$/', $addParams['ns_ip_address']) && preg_match('/^((25[0-5]|(2[0-4]|1[0-9]|[1-9]|)[0-9])(\.(?!$)|$)){4}$/', $addParams['ns_ip_address']))
-            && !(!preg_match('/^[0-9\.]+$/', $addParams['ns_ip_address']) && preg_match('/^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9]+)\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$/', $addParams['ns_ip_address']))
-        ) {
+        if (! filter_var($addParams['ns_ip_address'], FILTER_VALIDATE_DOMAIN)
```
Comment (sc979) [inline www/class/centreon-clapi/centreonInstance.class.php]:
```suggestion
        if (!filter_var($addParams['ns_ip_address'], FILTER_VALIDATE_DOMAIN)
```

File: www/class/centreon-clapi/centreonInstance.class.php [hunk 5]
```diff
@@ -110,10 +110,9 @@ public function initInsertParameters($parameters)
...
+            && !filter_var($addParams['ns_ip_address'], FILTER_VALIDATE_IP))
```
Comment (sc979) [inline www/class/centreon-clapi/centreonInstance.class.php]:
```suggestion
            && !filter_var($addParams['ns_ip_address'], FILTER_VALIDATE_IP)
```

File: www/class/centreon-clapi/centreonInstance.class.php [hunk 6]
```diff
@@ -110,10 +110,9 @@ public function initInsertParameters($parameters)
...
+        {
```
Comment (sc979) [inline www/class/centreon-clapi/centreonInstance.class.php]:
```suggestion
        ) {
```

File: www/include/configuration/configServers/DB-Func.php [hunk 7]
```diff
@@ -159,10 +159,7 @@ function testExistence($name = null): bool
 function isValidIpAddress($ipAddress): bool
 {
     // Check IPv6, IPv4 and FQDN format
-    if (!preg_match('/^([0-9a-fA-F]{4}|0)(\:([0-9a-fA-F]{4}|0)){7}$/', $ipAddress)
-        && !(preg_match('/^[0-9\.]+$/', $ipAddress) && preg_match('/^((25[0-5]|(2[0-4]|1[0-9]|[1-9]|)[0-9])(\.(?!$)|$)){4}$/', $ipAddress))
-        && !(!preg_match('/^[0-9\.]+$/', $ipAddress) && preg_match('/^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9]+)\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$/', $ipAddress))
-    ) {
+    if (! filter_var($ipAddress, FILTER_VALIDATE_DOMAIN) && !filter_var($ipAddress, FILTER_VALIDATE_IP)) {
```
Comment (sc979) [inline www/include/configuration/configServers/DB-Func.php]:
```suggestion
    if (!filter_var($ipAddress, FILTER_VALIDATE_DOMAIN) && !filter_var($ipAddress, FILTER_VALIDATE_IP)) {
```

File: www/class/centreon-clapi/centreonInstance.class.php [hunk 8]
```diff
@@ -107,6 +108,14 @@ public function initInsertParameters($parameters)
             throw new CentreonClapiException('Incorrect port parameters');
         }
         $addParams['ssh_port'] = $params[self::ORDER_SSH_PORT];
+
+        // Check IPv6, IPv4 and FQDN format
+        if (!filter_var($addParams['ns_ip_address'], FILTER_VALIDATE_DOMAIN)
```
Comment (sc979) [inline www/class/centreon-clapi/centreonInstance.class.php]:
```suggestion
        if (
            !filter_var($addParams['ns_ip_address'], FILTER_VALIDATE_DOMAIN)
```

File: www/class/centreon-clapi/centreonInstance.class.php [hunk 9]
```diff
@@ -126,6 +136,14 @@ public function initUpdateParameters($parameters)
             throw new CentreonClapiException(self::MISSINGPARAMETER);
         }
 
+        // Check IPv6, IPv4 and FQDN format
+        if ($params[1] == 'ns_ip_address' && !preg_match('/^([0-9a-fA-F]{4}|0)(\:([0-9a-fA-F]{4}|0)){7}$/', $params[2])
+            && !(preg_match('/^[0-9\.]+$/', $params[2]) && preg_match('/^((25[0-5]|(2[0-4]|1[0-9]|[1-9]|)[0-9])(\.(?!$)|$)){4}$/', $params[2]))
```
Comment (sc979) [inline www/class/centreon-clapi/centreonInstance.class.php]: is it possible to use filter_var on thes lines too ?

File: www/class/centreon-clapi/centreonInstance.class.php [hunk 10]
```diff
@@ -126,6 +136,13 @@ public function initUpdateParameters($parameters)
             throw new CentreonClapiException(self::MISSINGPARAMETER);
         }
 
+        // Check IPv6, IPv4 and FQDN format
+        if ($params[1] == 'ns_ip_address' && !filter_var($params[2], FILTER_VALIDATE_DOMAIN)
```
Comment (sc979) [inline www/class/centreon-clapi/centreonInstance.class.php]:
```suggestion
        if (
            $params[1] == 'ns_ip_address'
            && !filter_var($params[2], FILTER_VALIDATE_DOMAIN)
```

**Pull request merged**